addons/mail/static/src/components/discuss/discuss_tests.js

```diff
@@ -5519,6 +5519,303 @@ QUnit.test('moderation: sent message rejected in moderated channel', async funct
+        res_id: 20,
+    }];
+
+
+    await afterNextRender(() =>
+        document.querySelector(`
+            .o_DiscussSidebar_item[data-thread-local-id="mail.thread_mail.channel_20"]
+        `).click()
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Message',
+        "the pending moderation message should be in the channel"
+    );
+    assert.strictEqual(
+        document.querySelector('.o_Message_content').textContent,
+        "test",
+        "the pending moderation message should have the correct content"
+    );
+});
+
+QUnit.test('moderation: new pending moderation message posted by me should appear in origin thread', async function (assert) {
+    assert.expect(2);
+
+    const self = this;
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                id: 20,
+                channel_type: "channel",
+                name: "general",
+                moderation: true,
+            }],
+        },
+    });
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'message_post') {
+                // simulate receiving the message
+                const message = {
+                    author_id: [self.env.session.partner_id, "Me"],
+                    body: args.kwargs.body,
+                    channel_ids: [], // server do NOT return channel_id of the message if pending moderation
+                    moderation_status: 'pending_moderation',
+                    id: 2,
+                    message_type: args.kwargs.message_type,
+                    model: 'mail.channel',
+                    subtype_xmlid: args.kwargs.subtype_xmlid,
+                    record_name: 'General',
+                    res_id: 20,
+                };
+                const notifications = [[
+                    ['my-db', 'res.partner', self.env.session.partner_id],
+                    { type: 'author', message },
+                ]];
+                self.widget.call('bus_service', 'trigger', 'notification', notifications);
+            }
+            return this._super(...arguments);
+        }
+    });
+    await afterNextRender(() =>
+        document.querySelector(`
+            .o_DiscussSidebar_item[data-thread-local-id="mail.thread_mail.channel_20"]
+        `).click()
+    );
+
+    await afterNextRender(() => {
+        document.querySelector('.o_ComposerTextInput').focus();
+        document.execCommand('insertText', false, "Moderated test");
+    });
+    await afterNextRender(() =>
+        document.querySelector('.o_Composer_buttonSend').click()
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Message',
+        "the pending moderation message should be in the channel"
+    );
+    assert.strictEqual(
+        document.querySelector('.o_Message_content').textContent,
+        "Moderated test",
+        "the pending moderation message should have the correct content"
+    );
```

Comment:
Both assertions could be turned to a single `.o_Message[data-message-local-id="[...]"]` (check whole diff)